### PR TITLE
Fix PHP notices in hathor with new menu items

### PR DESCRIPTION
**Pull request by wilsonge**
Pull Request for Issue https://github.com/joomla/joomla-cms/pull/15079#issuecomment-291426073 .

### Summary of Changes
JM mentioned creating a menu item in hathor and then switching back to ISIS gave a PHP Notice. This should patch that case (note I couldn't reproduce so I'm not 100%)

Note this does not 'fix' adding admin menu items to hathor. it just fixes the bug about creating the site ones

### Testing Instructions
Check menu is created in hathor without warnings. Swap to isis and see no warnings

### Expected result
It all works

### Actual result
`Fatal error: Cannot use object of type stdClass as array in /administrator/components/com_menus/models/item.php on line 603`

### Documentation Changes Required
None

### CODE REVIEW

File: administrator/templates/hathor/html/com_menus/item/edit.php [hunk 1]
```diff
@@ -72,11 +72,17 @@
...
+$isModal  = $input->get('layout') == 'modal' ? true : false;
+$layout   = $isModal ? 'modal' : 'edit';
+$tmpl     = $isModal || $input->get('tmpl', '', 'cmd') === 'component' ? '&tmpl=component' : '';
+$clientId = $this->state->get('item.client_id', 0);
 ?>
 
 <div class="menuitem-edit">
 
-<form action="<?php echo JRoute::_('index.php?option=com_menus&layout=edit&id='.(int) $this->item->id); ?>" method="post" name="adminForm" id="item-form" class="form-validate">
+<form action="<?php echo JRoute::_('index.php?option=com_menus&view=item&client_id=' . $clientId . '&layout=' . $layout . $tmpl . '&id=' . (int) $this->item->id); ?>" method="post" name="adminForm" id="item-form" class="form-validate">
```
Comment (C-Lodder) [inline administrator/templates/hathor/html/com_menus/item/edit.php]: if `$tmpl === 'component'`, then the URL will end up being `&layout=modalcomponent`. Is that right?

Comment (wilsonge) [inline administrator/templates/hathor/html/com_menus/item/edit.php]: I don't think you can combine those things. But even if you can is that any different to isis right now? https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_menus/views/item/tmpl/edit.php#L91-L95

Comment (C-Lodder) [inline administrator/templates/hathor/html/com_menus/item/edit.php]: If they can't be combined, then why not just do:

```
$tmpl = $isModal || $input->get('tmpl', '', 'cmd') === '' ? '&tmpl=component' : '';
```
?

I assume `component` is there for a reason. If it needs to stay, it should be fixed in Isis and the fix ported over to Hathor

Comment (C-Lodder) [inline administrator/templates/hathor/html/com_menus/item/edit.php]: remind me never to look at PHP at night.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 1fc71e687a479c29cdbd078a372439107936a6cb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15118">issues.joomla.org/tracker/joomla-cms/15118</a>.</sub>

**Pull request merged**